### PR TITLE
Cupertino localization step 1: add an English arb file

### PR DESCRIPTION
**Pull request by xster**
## Description

Trying to figure out how to split this feature into steps. First, create the reference arb file to create the translation console xmls out of. 

## Related Issues

#13452

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes tests for *all* changed/updated/fixed behaviors (See [Test Coverage]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require Flutter developers to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (Please read [Handling breaking changes]).
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

File: dev/tools/localizations_utils.dart [hunk 1]
```diff
@@ -24,7 +24,10 @@ class LocaleInfo implements Comparable<LocaleInfo> {
   /// and country is 2-3 characters and all uppercase.
   ///
   /// 'language_COUNTRY' or 'language_script' are also valid. Missing fields will be null.
-  factory LocaleInfo.fromString(String locale, {bool assume = false}) {
+  ///
+  /// When `deriveScriptCode` is true, the [scriptCode] will be tentatively derived from
```
Comment (HansMuller) [inline dev/tools/localizations_utils.dart]: Tentatively? What would override the tentative value?

It looks like deriveScriptCode indices that the scriptCode should be computed from the locale string if it wasn't specified.

File: packages/flutter/lib/src/cupertino/localizations.dart [hunk 2]
```diff
@@ -92,6 +95,7 @@ abstract class CupertinoLocalizations {
   ///
   /// - US English: Wed Sep 27
   /// - Russian: ср сент. 27
+  // Based on intl package's DateFormat.MMMEd.
```
Comment (HansMuller) [inline packages/flutter/lib/src/cupertino/localizations.dart]: Just based on the other similar comments ... The global version is based on ...?

File: packages/flutter_localizations/lib/src/l10n/cupertino_en.arb [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+  },
+
+  "timerPickerHourLabelOne": "hour",
+  "timerPickerHourLabelOther": "hours",
+  "@timerPickerHourLabel": {
+    "description": "The label adjacent to an hour integer number in a countdown timer.",
+    "plural": "hour"
+  },
+
+  "timerPickerMinuteLabelOther": "min",
```
Comment (HansMuller) [inline packages/flutter_localizations/lib/src/l10n/cupertino_en.arb]: minute?
I think the one case is needed here

This one should probably be "$minute minutes"

File: packages/flutter_localizations/lib/src/l10n/cupertino_en.arb [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+    "description": "The abbreviation for post meridiem (after noon) shown in the time picker.",
+  },
+
+  "alertDialogLabel": "Alert",
+  "@alertDialogLabel": {
+    "description": "The accessibility audio announcement made when an iOS style alert dialog is opened."
+  },
+
+  "timerPickerHourLabelOne": "hour",
+  "timerPickerHourLabelOther": "hours",
```
Comment (HansMuller) [inline packages/flutter_localizations/lib/src/l10n/cupertino_en.arb]: expect to see $hour in the other translation, like "$hour hours"

File: packages/flutter_localizations/lib/src/l10n/cupertino_en.arb [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+    "plural": "hour"
+  },
+
+  "timerPickerMinuteLabelOther": "min",
+  "@timerPickerMinuteLabel": {
+    "description": "The label adjacent to a minute integer number in a countdown timer.",
+    "plural": "minute"
+  },
+
+  "timerPickerSecondLabelOther": "sec",
```
Comment (HansMuller) [inline packages/flutter_localizations/lib/src/l10n/cupertino_en.arb]: Same feedback as before about plural translations

File: packages/flutter_localizations/lib/src/l10n/cupertino_en.arb [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+  },
+
+  "datePickerMinuteSemanticsOne": "1 minute",
+  "datePickerMinuteSemanticsOther": "$minute minutes",
+  "@datePickerMinuteSemanticsLabel": {
+    "description": "Accessibility announcement for the selected minute on a time picker such as '15 minutes' or '15分'",
+    "plural": "minute"
+  },
+
+  "datePickerDateOrder": "mdy",
```
Comment (HansMuller) [inline packages/flutter_localizations/lib/src/l10n/cupertino_en.arb]: Do we really use abbreviations in enums like this? Flutter is typically monthDayYear

**Review by HansMuller — APPROVED** [review]
LGTM, the English plural translations need some work

File: dev/tools/localizations_utils.dart (hunk 1 above)
Comment (xster) [reply dev/tools/localizations_utils.dart]: you're right. This wasn't very clear. Reworded

File: packages/flutter/lib/src/cupertino/localizations.dart (hunk 2 above)
Comment (xster) [reply packages/flutter/lib/src/cupertino/localizations.dart]: True. Changed

File: packages/flutter_localizations/lib/src/l10n/cupertino_en.arb (hunk 3 above)
Comment (xster) [reply packages/flutter_localizations/lib/src/l10n/cupertino_en.arb]: so this one is a bit weird... it's not actually part of the same string as $minutes. See https://flutter.dev/images/widget-catalog/cupertino-timer-picker.png. It's in a separate widget. Its pluralization depends on its sibling widget but it's not one string.

File: packages/flutter_localizations/lib/src/l10n/cupertino_en.arb (hunk 4 above)
Comment (xster) [reply packages/flutter_localizations/lib/src/l10n/cupertino_en.arb]: same

File: packages/flutter_localizations/lib/src/l10n/cupertino_en.arb (hunk 5 above)
Comment (xster) [reply packages/flutter_localizations/lib/src/l10n/cupertino_en.arb]: same

File: packages/flutter_localizations/lib/src/l10n/cupertino_en.arb (hunk 3 above)
Comment (xster) [reply packages/flutter_localizations/lib/src/l10n/cupertino_en.arb]: Does it? According to localizations_validator.dart, only Other is needed.

File: packages/flutter_localizations/lib/src/l10n/cupertino_en.arb (hunk 6 above)
Comment (xster) [reply packages/flutter_localizations/lib/src/l10n/cupertino_en.arb]: This is referring to https://github.com/xster/flutter/blob/9da3df5ba4e4cac46620e153bdf972ebde25bd58/packages/flutter/lib/src/cupertino/localizations.dart#L34. What were you referring to in Flutter as typical? Maybe we shouldn't have created that enum if an existing one's use is already widespread... Though that's a breaking API>

Comment (HansMuller) [reply packages/flutter_localizations/lib/src/l10n/cupertino_en.arb]: I didn't mean that there was an existing enum with similar semantics, I was just surprised by the abbreviation. Water under the bridge now :-)

File: packages/flutter_localizations/lib/src/l10n/cupertino_en.arb (hunk 3 above)
Comment (HansMuller) [reply packages/flutter_localizations/lib/src/l10n/cupertino_en.arb]: Thanks for the pointer the screenshot, I didn't understand how this was going to be used. You're right that only the "Other" plural translation is needed. It's going to be tricky to explain what's needed to the translators however it's possible to include a screenshot. That will help.

Comment (xster) [reply packages/flutter_localizations/lib/src/l10n/cupertino_en.arb]: ya, I saw the screenshot mechanism on the console. I'll likely add all of them to remind them of the iOS context.

**Pull request merged**